parse_node/parse_class_declaration.ts

```diff
@@ -140,6 +154,29 @@ class Foo {
 }`,
   expected: `
 class_name Foo
-var x: int = 1  
+var x: int = 1
+`,
+}
+
+export const testExportArgsSetGet: Test = {
+  ts: `
+@autoload
+class Foo {
+  @export_args(PackedScene)
```

Comment:
Could we not make @exports take optional arguments, so we don't introduce another keyword?
Additionally wouldn't we be able to infer the type here instead of needing to manually provide it?
Afaict the codebase tried to avoid manual typing.

Reply:
👍 to this comment. @exports taking optional arguments would be the ideal here!

Reply:
Do we care about backwards compatibility? Because if we add arguments (even optional) to `exports` we cannot use it as we used to. 

```ts
// exports has no arguments, currently this code is valid
export class X extends Node2D {
  @exports
  test: int = 0;
}

// when exports definition is changed to accept arguments we need to add () to the end 
// of exports to make TypeScript not complain, this is not backward compatible
export class X extends Node2D {
  @exports()
  test: int = 0;
}
```

Also about infering the type, we can do that for array and maps but what about other cases like `DIR`, `MULTILINE`, `RGB`, etc. ?

Reply:
> Do we care about backwards compatibility? Because if we add arguments (even optional) to exports we cannot use it as we used to.

Actually, I just tested it and if you make exports definition `declare function exports(...args: any[]): (target: Node, name: string) => void` then TypeScript does not care if it is really invoked or not. That means current code will also work. Sorry for misleading :)

Reply:
I actually didn't use DIR, Multiline etc. yet. That would indeed be more complicated.
So if no arg is provided, we infer and otherwise we use what's provided?

Reply:
I've decided to infer and also add provided arguments after the inferred ones. See newest commits :)